cmd/snap/cmd_run.go

```diff
@@ -227,6 +230,141 @@ func isReexeced() bool {
 	return strings.HasPrefix(exe, dirs.SnapMountDir)
 }
 
+func migrateXauthority(info *snap.Info) (string, error) {
+	u, err := userCurrent()
+	if err != nil {
+		return "", fmt.Errorf(i18n.G("cannot get the current user: %s"), err)
+	}
+
+	// If our target directory (XDG_RUNTIME_DIR) doesn't exist we
+	// don't attempt to create it.
+	baseTargetDir := filepath.Join(dirs.XdgRuntimeDirBase, u.Uid)
+	if !osutil.FileExists(baseTargetDir) {
+		return "", nil
+	}
+
+	xauthPath := osGetenv("XAUTHORITY")
+	if len(xauthPath) == 0 || !osutil.FileExists(xauthPath) {
+		// Nothing to do for us. Most likely running outside of any
+		// graphical X11 session.
+		return "", nil
+	}
+
+	fin, err := os.Open(xauthPath)
+	if err != nil {
+		return "", err
```

Comment:
Message needs to point out that this is about the Xauthority file, as the filename might not give a hint. So user ends up with a "cannot open" message without knowing why it's even trying to open.